### ISSUE TITLE
2.0.1 - Aspect Ratio gets messed up using Metal rendering.

### ISSUE THREAD

**Issue by dineshflock** [twilio/video-quickstart-swift#266]
### Description

> Related to help desk request (1360773)

**Problem**
With the the reusability change in the TVIVideoView (2.0.0) 
1. Some times on scrolling the cells the aspect ratio gets messed up.
2. Say I have just the local participant in the room and its video track is being rendered in a collection view cell (code below). When I go in background and and then in foreground, the mirroring gets reset along with above problem.

[Reference image](https://imgur.com/a/RaaEP1x)

**Questions**
On the help desk we got this response

> The videoDimensions, videoOrientation, viewShouldRotateContent and mirror properties are reset to their respective default values.
> 
> Can you verify that you're explicitly setting the values?

1. When do these properties gets reset? On adding this view as renderer to a track?
2. Apart from `mirror`, how do we exactly set these properties?
3. A sample code snippet how to approach this would really help us.
#### Code

```
@interface VideoCollectionViewCell() <TVIVideoViewDelegate>

@property (nonatomic) NSString *uniqueIdentifier;

@property (nonatomic, weak) IBOutlet UIView *borderView;
@property (nonatomic, weak) IBOutlet TVIVideoView *videoView;
@property (nonatomic, weak) IBOutlet UIImageView *profileImageView;
@property (nonatomic, weak) IBOutlet UIImageView *muteIndicatorImageView;

@property (nonatomic, weak) TVIVideoTrack *videoTrackBeingRendered;

@end

@implementation VideoCollectionViewCell

- (void)awakeFromNib {
  [super awakeFromNib];
  [self configureBorderView];
  self.videoView.delegate = self;
  self.muteIndicatorImageView.layer.masksToBounds = YES;
  self.muteIndicatorImageView.layer.cornerRadius = CGRectGetHeight(self.muteIndicatorImageView.frame) / 2;
  [self setDefaults];
}

- (void)setVideoSelected:(BOOL)videoSelected {
  _videoSelected = videoSelected;
  self.borderView.hidden = !_videoSelected;
}

- (void)configureBorderView {
  self.borderView.layer.borderWidth = BorderWidth;
  self.borderView.layer.borderColor = [UIColor colorWithIntegerValue:BorderColor alpha:1.0].CGColor;
}

- (void)prepareForReuse {
  [super prepareForReuse];
  [self setDefaults];
}

- (void)setDefaults {
  [self stopRenderingTrack];
  [self setVideoSelected:NO];
  [self setUniqueIdentifier:nil];
  [self setMuted:NO];
  [self setAvatar:nil];
  [self setVideoViewMirroring:NO];
  [self showAvatar];
}

- (void)setAvatar:(UIImage *)image {
  [self.profileImageView setImage:image];
}

- (void)setMuted:(BOOL)isMuted {
  self.muteIndicatorImageView.hidden = !isMuted;
}

- (void)setVideoViewMirroring:(BOOL)mirror {
  self.videoView.mirror = mirror;
}

- (void)showAvatar {
  [self.videoView setHidden:YES];
}

- (void)showVideoViewIfAvailable {
  if (self.videoView.hasVideoData) {
    self.videoView.contentMode = UIViewContentModeScaleAspectFill;
    [self.videoView setHidden:NO];
  }
}

- (void)renderTrack:(TVIVideoTrack *)track {
  [self stopRenderingTrack];
  [track.renderers containsObject:self.videoView] ?: [track addRenderer:self.videoView];
  self.videoTrackBeingRendered = track;
}

-(void)stopRenderingTrack {
  [self.videoTrackBeingRendered removeRenderer:self.videoView];
}

#pragma mark - Convinience methods for setting up cell

- (void)setupForParticipant:(VideoCallParticipant *)participant {
  ParameterAssert(participant);
  [self setUniqueIdentifier:participant.uniqueIdentifier];
  [self setMuted:participant.isMuted];
  [self setVideoViewMirroring:participant.shouldMirrorVideoWhenDisplaying];
  __weak typeof(self) weakSelf = self;
  [participant avatarWithCompletionBlock:^(UIImage *image, NSError *error) {
    __strong typeof(self) strongSelf = weakSelf;
    if ([strongSelf.uniqueIdentifier isEqualToString:participant.uniqueIdentifier]) {
      [strongSelf setAvatar:image];
    }
  }];
  [self renderTrack:participant.videoTrack];
  if (participant.videoEnabled) {
    [self showVideoViewIfAvailable];
  } else {
    [self showAvatar];
  }
}

- (void)unhideVideoViewIfNeeded {
  if (self.videoView.hidden && self.videoTrackBeingRendered.enabled) {
    self.videoView.contentMode = UIViewContentModeScaleAspectFill;
    self.videoView.hidden = NO;
  }
}

- (void)updateForMuteStateChangeForParticipant:(VideoCallParticipant *)participant {
  ParameterAssert(participant);
  [self setMuted:participant.isMuted];
}

- (void)updateForVideoStateChangeForParticipant:(VideoCallParticipant *)participant
                            shouldReRenderTrack:(BOOL)shouldReRenderTrack {
  ParameterAssert(participant);
  shouldReRenderTrack ? [self renderTrack:participant.videoTrack] : nil;
  participant.isVideoEnabled ? [self showVideoViewIfAvailable] : [self showAvatar];
}

#pragma mark - TVIVideoView delegate methods

- (void)videoViewDidReceiveData:(TVIVideoView *)view {
  [self unhideVideoViewIfNeeded];
}

@end
```

#### Reproduces How Often

50%

### Versions

All relevant version information for the issue.

#### Video iOS SDK

2.0.1 via CocoaPods

#### Xcode

9.2

Thanks
Dinesh

**Comment by ceaglest** [comment]
Hey @dineshflock,

This bug is fixed in our [2.1.0](https://www.twilio.com/docs/admin/pages/3466/edit/preview/#210-may-24-2018) release, please give it a shot.

We're very sorry about the regression and are taking steps to improve our test coverage in this area.

Best,
Chris

**Comment by ceaglest** [comment]
Closing now that 2.1.0 has been available for some time. Please let us know if you're still having any issues in this area.
